ank/doc/swdesign/README.md

```diff
@@ -756,6 +758,26 @@ Needs:
 - utest
 - stest
 
+### Handling other message while waiting for response
+
+|[Store unexpected messages](plantuml/seq_store_missed_messages.svg)
+
+#### CLI stores unexpected messages
+`swdd~cli-stores-unexpected-message~1`
+
+Status: approved
+
+While the Ankaios CLI waits for a response from the Ankaios Server,
+the Ankaios CLI should store all other received messages.
+
+Rationale:
+The Ank CLI needs these message to watch changed workloads,
+as the first `UpdateWorkloadState` message can already be received while waiting for the response the `UpdateStateRequest` and the `CompleteStateRequest`.
```

Comment:
Weak requirement, please use `shall`. 
Additionally the problem is not completely described in the rationale. Maybe:
```suggestion
While the Ankaios CLI waits for a response from the Ankaios Server,
the Ankaios CLI shall store all unrelated received messages for later processing.

Rationale:
While communicating with the Ankaios server the ank CLI could already receive unsolicited `UpdateWorkloadState` messages that are needed later during command execution. Storing the messages ensure that they are not missed and the CLI does not block endlessly waiting for them, e.g., while waiting for the end state for an apply command.
```